app/assets/scripts/views/field-report-form.js

```diff
@@ -0,0 +1,1235 @@
+            checked: false
+          })),
+          description: undefined
+        },
+        actionsFederation: {
+          options: formData.actions.map(o => ({
+            name: o.name,
+            checked: false
+          })),
+          description: undefined
+        },
+
+        // Step 4
+        dref: undefined,
+        amountDref: undefined,
+        emergencyAppeal: undefined,
+        amountEmergencyAppeal: undefined,
+        rdrtrits: undefined,
+        numPplRdrits: undefined,
+        fact: undefined,
+        numPplFact: undefined,
+        ifrcStaff: undefined,
+        numPplIfrcStaff: undefined,
+
+        // Step 5
+        contactOriginator: { name: undefined, func: undefined, email: undefined },
+        contactPrimary: { name: undefined, func: undefined, email: undefined },
+        contactNatSoc: { name: undefined, func: undefined, email: undefined },
+        contactFederation: { name: undefined, func: undefined, email: undefined },
+        contactMediaNatSoc: { name: undefined, func: undefined, email: undefined },
+        contactMedia: { name: undefined, func: undefined, email: undefined }
+      },
+      errors: null
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onStepBackClick = this.onStepBackClick.bind(this);
+  }
+
+  onSubmit (e) {
+    e.preventDefault();
+    const { step, data } = this.state;
+    let state = prepStateForValidation(data);
+    console.log('state', state);
+
+    let validator;
+    switch (step) {
+      case 1:
+        validator = ajv.compile(schemaStep1);
+        break;
+      case 2:
+        validator = ajv.compile(schemaStep2);
+        break;
+      case 3:
+        // Schema is being compiled and used but there are no errors being
+        // displayed for this step, for no fields are required nor prone
+        // to error.
+        validator = ajv.compile(schemaStep3);
+        break;
+      case 4:
+        validator = ajv.compile(schemaStep4);
+        break;
+      case 5:
+        validator = ajv.compile(schemaStep5);
+        break;
+    }
+    validator(state);
+
+    this.setState({ errors: _cloneDeep(validator.errors) });
+    console.log('validator.errors', validator.errors);
+    if (validator.errors !== null) {
+      return null;
+    } else {
+      console.log('good');
+      return;
+      if (step === 5) {
+        console.log('Submit data!!!');
+      } else {
+        this.setState({ step: step + 1 });
+      }
+    }
+  }
+
+  onFieldChange (field, e) {
+    let data = _cloneDeep(this.state.data);
+    let val = e.target ? e.target.value : e;
+    _set(data, field, val === '' ? undefined : val);
+    this.setState({data});
+  }
+
+  onStepBackClick () {
+    if (this.state.step > 1) {
+      this.setState({ step: this.state.step - 1 });
+    }
+  }
+
+  onStepperClick (step, e) {
+    e.preventDefault();
+    // Can't move forward. Only backwards.
+    // if (step > this.state.step) {
+    //   return;
+    // }
```

Comment:
This is probably Ok to leave out completely yeah? People will be able to hop around if they wish.

Reply:
This is a behavior we have to decide on. Even if hopping around we should ensure that the page they're in is valid. They shouldn't be able to move around if there are errors in the page. no?